File: CS/DevExpress.XtraReports.CustomControls/XRGridDesigner.cs
using DevExpress.XtraReports.Design;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.Design;
using System.Drawing.Design;
using System.Linq;
using System.Text;

namespace DevExpress.XtraReports.CustomControls
{
    public class XRGridDesigner : XRDataContainerControlDesigner
    {
        public XRGridDesigner() : base() { }

        protected override XRDataContainerDesignerColumnActionList CreateColumnActionList()
        {
            return new XRGridColumnActionList(this);
        }
    }

    public class XRGridColumnActionList : XRDataContainerDesignerColumnActionList
    {
        public XRGridColumnActionList(XRComponentDesigner componentDesigner)
            : base(componentDesigner)
        {
        }

        protected override void AddHeadersPropertyItem(DesignerActionItemCollection actionItems)
        {
            AddPropertyItem(actionItems, control.FieldHeaderName, "Columns");
        }

        [Editor(typeof(XRCollectionEditor), typeof(UITypeEditor))]
        public XRGridColumnCollection Columns
        {
            get
            {
                return base.Headers as XRGridColumnCollection;
            }
        }
    }
}

File: VB/DevExpress.XtraReports.CustomControls/XRGridDesigner.vb
Imports DevExpress.XtraReports.Design
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.ComponentModel.Design
Imports System.Drawing.Design
Imports System.Linq
Imports System.Text

Namespace DevExpress.XtraReports.CustomControls
	Public Class XRGridDesigner
		Inherits XRDataContainerControlDesigner

		Public Sub New()
			MyBase.New()
		End Sub

		Protected Overrides Function CreateColumnActionList() As XRDataContainerDesignerColumnActionList
			Return New XRGridColumnActionList(Me)
		End Function
	End Class

	Public Class XRGridColumnActionList
		Inherits XRDataContainerDesignerColumnActionList

		Public Sub New(ByVal componentDesigner As XRComponentDesigner)
			MyBase.New(componentDesigner)
		End Sub

		Protected Overrides Sub AddHeadersPropertyItem(ByVal actionItems As DesignerActionItemCollection)
			AddPropertyItem(actionItems, control.FieldHeaderName, "Columns")
		End Sub

		<Editor(GetType(XRCollectionEditor), GetType(UITypeEditor))>
		Public ReadOnly Property Columns() As XRGridColumnCollection
			Get
				Return TryCast(MyBase.Headers, XRGridColumnCollection)
			End Get
		End Property
	End Class
End Namespace
